src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -2068,7 +2068,12 @@ void M1(C c)
 }
 ");
             c.VerifyDiagnostics(
-                );
+                // (13,17): warning CS8602: Dereference of a possibly null reference.
```

Comment:
>// (13,17): warning CS8602: Dereference of a possibly null reference. [](start = 16, length = 69)

Description of the PR sounds like a pure optimization, but there is a behavior change. Was this behavior change one of the goals of the PR, or is this just a fall out of the change that are willing to live with?

Reply:
The old behavior for this scenario doesn't seem much better than the new behavior, and not worth the cost.